docs/rules/no-status-code-assert.md

```diff
@@ -0,0 +1,15 @@
+# Disallow the use of `statusCode` as a parameter in assert.
+
+## Fail
+
+```js
+assert(statusCode === 200);
```

Comment:
update to reflect the latest implementation?

Reply:
@carlansley should we have rule to fix the prod code with statusCode if we have numbers?

Reply:
Not sure I can answer that, can we add a little more description to the issue/PR about what this entire rule is needed for?  I'm sure this was discussed at some point but tbh I don't recall what the original problem was.

Reply:
We need to monitor for assertion failures in production and also improve how assertions are used in production code. Currently, assertions are being used for status checks, which isn’t ideal. A good first step is to implement an ESLint rule that will detect when assertions are made using a statusCode as one of the parameters. This will help identify and clean up improper use of assertions in the code.  notify service could look for assertion errors, but we need to clean up usage of assertions for status checks.  

https://checkdigit.slack.com/archives/C0W7RDCE4/p1732226670820699?thread_ts=1732223039.438189&cid=C0W7RDCE4

As we are checking for statusCode,  should we have another rule to auto fix the prod code with statusCode if we have numbers some thing like 
assert(statusCode === 200);
 would be assert(statusCode === StatusCodes.OK);

Reply:
is there a mistake in the documentation wrt what is "pass" and what is "fail"?  That was what confused me initially.

Reply:
But initially it was like ## Fail

```js
assert(statusCode === 200);
assert.equal(statusCode, 200);

Updated the rule to handle just statusCodes.

## Fail

```js
assert(statusCode === StatusCodes.OK);
assert.equal(statusCode, StatusCodes.OK);
```

## Pass

```js
assert(response.status === 200);
assert.equal(response.status, 200);
```

Pass wouldn't report as StatusCodes aren't used.  This was the reason if we need to have another rule to auto fix all the numbers to statusCodes?